### PR TITLE
bugfix/IOS-2192-crash-on-mqtt-reconfirgure

### PR DESCRIPTION
**Pull request by ericgimenezgalera**
- Check if is nil some parameters when we trying to disconnect

### CODE REVIEW

**Review by mmarfull — APPROVED** [review]

File: MQTTClient/MQTTClient/MQTTSession.m [hunk 1]
```diff
@@ -548,15 +548,17 @@ - (void)closeInternal {
...
-    NSArray *flows = [self.persistence allFlowsforClientId:self.clientId
-                                              incomingFlag:NO];
-    for (id<MQTTFlow> flow in flows) {
-        switch ((flow.commandType).intValue) {
-            case MQTTPublish:
-            case MQTTPubrel:
-                flow.deadline = [flow.deadline dateByAddingTimeInterval:-self.dupTimeout];
-                [self.persistence sync];
-                break;
+    if (self.clientId) {
```
Comment (mmarfull) [inline MQTTClient/MQTTClient/MQTTSession.m:551]: What happens if clientId doesn't have value.We want to know it?

**Review by mmarfull — APPROVED** [review]

File: MQTTClient/MQTTClient/MQTTSession.m [hunk 2]
```diff
@@ -572,24 +574,28 @@ - (void)closeInternal {
...
+    if(self.unsubscribeHandlers) {
+        NSArray *allUnsubscribeHandlers = self.unsubscribeHandlers.allValues;
+        [self.unsubscribeHandlers removeAllObjects];
+        for (MQTTUnsubscribeHandler unsubscribeHandler in allUnsubscribeHandlers) {
+            unsubscribeHandler(error);
+        }
     }
 
     MQTTDisconnectHandler disconnectHandler = self.disconnectHandler;
     if (disconnectHandler) {
```
Comment (didix21) [inline MQTTClient/MQTTClient/MQTTSession.m:594]: Why no check here if disconnetHandler is != nil

Comment (ericgimenezgalera) [reply MQTTClient/MQTTClient/MQTTSession.m:594]: it's irrelevant

Comment (didix21) [reply MQTTClient/MQTTClient/MQTTSession.m:594]: then if it's irrelevant why do you need to assign self.disconnectHandler = nil outside of this check?

**Pull request merged**